source/extensions/access_loggers/wasm/wasm_access_log_impl.h

```diff
@@ -30,8 +32,8 @@ class WasmAccessLog : public AccessLog::Instance {
       }
     }
 
-    if (tls_slot_->get()) {
-      tls_slot_->getTyped<WasmHandle>().wasm()->log(root_id_, request_headers, response_headers,
+    if (tls_slot_->currentThreadRegistered() && tls_slot_->get()) {
```

Comment:
Can you add some comments about how we would get here with the current thread not being registered?

Reply:
Right, I don't think it can happen here. I was overzealous when adding this check to `runOnAllThreads()` calls.